otherlibs/win32unix/socket.c

```diff
@@ -24,17 +25,54 @@ int socket_type_table[] = {
   SOCK_STREAM, SOCK_DGRAM, SOCK_RAW, SOCK_SEQPACKET
 };
 
-CAMLprim value unix_socket(value cloexec, value domain, value type, value proto)
+SOCKET win32_socket(int domain, int type, int protocol,
+                    LPWSAPROTOCOL_INFO info,
+                    BOOL inherit)
 {
   SOCKET s;
+  DWORD flags = WSA_FLAG_OVERLAPPED;
```

Comment:
Do I understand well that `socket` is roughly equivalent to `WSASocket` + `WSA_FLAG_OVERLAPPED`?

Reply:
Yes. The [`socket()` doc](https://docs.microsoft.com/en-us/windows/win32/api/winsock2/nf-winsock2-socket) reads:
> The socket that is created will have the overlapped attribute as a default.